packages/language-server/src/worker.ts

```diff
@@ -0,0 +1,9 @@
+import { doSemanticAnalysis } from '@neo4j-cypher/language-support';
+import { parentPort } from 'worker_threads';
+
+parentPort.on('message', (value: { query: string; port: MessagePort }) => {
+  const result = doSemanticAnalysis(value.query);
```

Comment:
It's not a great name, do you have any other suggestions? Maybe something like `getSemanticDiagnostics` or simliar ?